src/hooks/screen/useChatScreenHook.ts

```diff
@@ -132,32 +155,48 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       const randomId = uuid();
 
       if (currentChatSchema === null) {
-        currentChatSchema = await ChatSchema.generateSendingChat(
-          randomId,
-          userId,
-          selectedChannel?.id,
-          message,
-          attachments,
-          localDb,
-          'regular',
-          'pending'
-        );
-        currentChatSchema.save(localDb);
-        if (selectedChannel) {
-          const channelList: ChannelList | null = await ChannelList.getSchemaById(
-            localDb,
-            selectedChannel?.id
-          );
-          if (channelList) {
-            channelList.description = message;
-            channelList.lastUpdatedBy = userId;
-            channelList.lastUpdatedAt = new Date().toISOString();
-            await channelList.save(localDb);
+        queue.addPriorityJob({
+          operationLabel: DatabaseOperationLabel.ChatScreen_SendChat,
+          id: `${selectedChannel?.id}-${new Date().valueOf()}`,
+          priority: QueueJobPriority.HIGH,
+          forceAddToQueue: true,
+          task: async () => {
+            currentChatSchema = await ChatSchema.generateSendingChat(
+              randomId,
+              userId,
+              selectedChannel?.id,
+              message,
+              attachments,
+              localDb,
+              'regular',
+              'pending'
+            );
+
+            await currentChatSchema?.save(localDb);
+            initChatData();
           }
-        }
+        });
 
-        refresh('chat');
-        refresh('channelList');
+        if (selectedChannel) {
+          queue.addPriorityJob({
+            operationLabel: DatabaseOperationLabel.ChatScreen_UpdateChannelDescription,
+            id: `${selectedChannel?.id}-${randomId}`,
+            priority: QueueJobPriority.HIGH,
+            task: async () => {
+              const channelList: ChannelList | null = await ChannelList.getSchemaById(
+                localDb,
+                selectedChannel?.id
+              );
+              if (channelList) {
+                channelList.description = message;
+                channelList.lastUpdatedBy = userId;
+                channelList.lastUpdatedAt = new Date().toISOString();
+                await channelList.save(localDb);
+                refresh('channelList');
+              }
+            }
+          });
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of the `sendChat` function is a significant improvement over the previous one. It uses the queue to manage database operations, which can enhance performance and maintainability. However, there's no error handling for the tasks added to the queue. If an error occurs during the execution of these tasks, it might lead to unexpected behavior or crashes. Consider adding error handling within each task in the queue.
```diff
        queue.addPriorityJob({
          operationLabel: DatabaseOperationLabel.ChatScreen_SendChat,
          id: `${selectedChannel?.id}-${new Date().valueOf()}`,
          priority: QueueJobPriority.HIGH,
          forceAddToQueue: true,
          task: async () => {
            try {
              currentChatSchema = await ChatSchema.generateSendingChat(
                randomId,
                userId,
                selectedChannel?.id,
                message,
                attachments,
                localDb,
                'regular',
                'pending'
              );
 
              await currentChatSchema?.save(localDb);
              initChatData();
            } catch (e) {
              console.error('Error while sending chat:', e);
            }
          }
        });
 
        if (selectedChannel) {
          queue.addPriorityJob({
            operationLabel: DatabaseOperationLabel.ChatScreen_UpdateChannelDescription,
            id: `${selectedChannel?.id}-${randomId}`,
            priority: QueueJobPriority.HIGH,
            task: async () => {
              try {
                const channelList: ChannelList | null = await ChannelList.getSchemaById(
                  localDb,
                  selectedChannel?.id
                );
                if (channelList) {
                  channelList.description = message;
                  channelList.lastUpdatedBy = userId;
                  channelList.lastUpdatedAt = new Date().toISOString();
                  await channelList.save(localDb);
                  refresh('channelList');
                }
              } catch (e) {
                console.error('Error while updating channel description:', e);
              }
            }
          });
        }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->